### PR TITLE
Flatten create operators

### PR DESCRIPTION
**Pull request by fickludd**
Welcome to this full-stack cypher experience. 

This change greatly improves performance for queries of the type

 `CREATE (a1)-[:R]->(b2), (a2)-[:R]->(a3),...,(a3000)-[:R]->(a3001)`

Which would otherwise often give stack overflows in all kinds of
places. The implemented solution replaces the previous nested
CreateNode and CreateRelationship operators with one Create operator,
that in turn has two flat lists of CreateNode and CreateRelationship
commands.

To also help for queries of consecutive CREATE clauses, these are now flattened into one.

 `CREATE (a) CREATE (b)` => `CREATE (a),(b)`

I also took the opportunity to simplify the Create pipes, and reuse
some more interpreted functionality in slotted.

### CODE REVIEW

**Comment by pontusmelke** [comment]
There are very many test failures, I will wait with reviewing until the number of failures drops to a more reasonable number :)

**Comment by fickludd** [comment]
@pontusmelke I messed up CypherComparisonSupport. Should be fixed now.

File: community/cypher/cypher-planner-3.5/src/main/scala/org/neo4j/cypher/internal/compiler/v3_5/ast/convert/plannerQuery/ClauseConverters.scala [hunk 1]
```diff
@@ -116,89 +117,96 @@ object ClauseConverters {
...
     }
 
-  private def addCreateToLogicalPlanInput(builder: PlannerQueryBuilder, clause: Create): PlannerQueryBuilder =
-    clause.pattern.patternParts.foldLeft(builder) {
+  private def addCreateToLogicalPlanInput(builder: PlannerQueryBuilder, clause: Create): PlannerQueryBuilder = {
+
+    val nodes = new ArrayBuffer[CreateNode]()
+    val relationships = new ArrayBuffer[CreateRelationship]()
+
+    // We need this locally to avoid creating node twice is they occur
```
Comment (pontusmelke) [inline community/cypher/cypher-planner-3.5/src/main/scala/org/neo4j/cypher/internal/compiler/v3_5/ast/convert/plannerQuery/ClauseConverters.scala:125]: This sentence is weird, `as they occur`?

File: community/cypher/cypher-planner-3.5/src/main/scala/org/neo4j/cypher/internal/compiler/v3_5/ast/convert/plannerQuery/StatementConverters.scala [hunk 2]
```diff
@@ -77,4 +77,35 @@ object StatementConverters {
...
+          prevCreate = None
+
+        case (nonCreate, None) =>
+          builder += nonCreate
+      }
+    }
+    for ((prevParts, pos) <- prevCreate)
+      builder += Create(Pattern(prevParts)(pos))(pos)
+    builder.result()
+//    clauses
```
Comment (pontusmelke) [inline community/cypher/cypher-planner-3.5/src/main/scala/org/neo4j/cypher/internal/compiler/v3_5/ast/convert/plannerQuery/StatementConverters.scala:109]: we can remove this

**Review by pontusmelke — CHANGES_REQUESTED** [review]
This looks good, there are still some failing tests though.

**Comment by fickludd** [comment]
Migrated to new repo. (and fixed review comments)

**Pull request closed without merging**